packages/flutter_tools/lib/src/commands/daemon.dart

```diff
@@ -417,7 +417,7 @@ class AppDomain extends Domain {
         );
         _sendAppEvent(app, 'stop');
       } catch (error) {
-        _sendAppEvent(app, 'stop', <String, dynamic>{'error': error.toString()});
+        _sendAppEvent(app, 'stop', <String, dynamic>{'error': _toJsonable(error)});
```

Comment:
This will send either a string, list, map, ToolExit, ... We should make this more uniform for the sake of clients.

Perhaps always send a `{'type':..., 'message': ..., 'exitCode':...}` map? If it's a `ToolExit` object, 'message' can be populated with obj.message, else it would be populated w/ `'$error'`. How does that sound?